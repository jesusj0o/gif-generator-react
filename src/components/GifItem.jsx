import propTypes from 'prop-types';

const GifItem = ({title, url, id}) => { 

  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: propTypes.string.isRequired, 
  url: propTypes.string.isRequired
}

export default GifItem;