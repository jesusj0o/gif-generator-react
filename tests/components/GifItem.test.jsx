import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Test en <GifItem />', () => {
    const title = "star wars"; 
    const url = "https://star-wars.com/yoda.jpg";
    test('should math with its snapshot', () => {
       const { container } = render(<GifItem title={title} url={url} />);
       expect(container).toMatchSnapshot();
    });

    test('should show the image with the indicated URL and ALT ', () => {
        render(<GifItem title={title} url={url}/>); 
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url); 
        expect(alt).toBe(title);
    });

    test('should show the title in a component', () => {
        render(<GifItem title={title} url={url} />); 
        expect(screen.getByText(title)).toBeTruthy()
    });
});