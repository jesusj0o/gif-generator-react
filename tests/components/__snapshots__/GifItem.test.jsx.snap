// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem /> should math with its snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="star wars"
      src="https://star-wars.com/yoda.jpg"
    />
    <p>
      star wars
    </p>
  </div>
</div>
`;
